components/fileselect/initial-files.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Initial Files
+page_title: FileSelect Initial Files
+description: The FileSelect allows you to define a collection of pre-selected files, which will be initially shown in the selected file list.
+slug: fileselect-initial-files
+tags: telerik,blazor,fileselect,initial, files
+published: True
+position: 40
+---
+
+# FileSelect Initial Files
+
+The `Files` parameter of the FileSelect accepts an `IEnumerable<FileSelectFileInfo>` collection that stores a set of pre-selected files. It allows you to display files in the file list initially, making it convenient for showcasing previously uploaded files.
+
+>caption Display initial files in FileSelect's list.
+
+```CSHTML
+
+<TelerikFileSelect Files="@InitialFiles" />
+
+@code {
+    private List<FileSelectFileInfo> InitialFiles { get; set; } = new List<FileSelectFileInfo>()
+    {
+        new FileSelectFileInfo(){ Id="1", Name="Report", Extension=".pdf", Size = 1024 * 1024 * 2 },
+        new FileSelectFileInfo(){ Id="2", Name="Image", Extension=".jpg", Size = 1024 * 1024 * 4 },
+        new FileSelectFileInfo(){ Id="3", Name="Picture", Extension=".png", Size = 1024 * 1024 * 3 },
+    };
+}
+
+```
+
+## Persist Selected Files
+
+The Initial Files feature of the FileSelect allows you to save the selected files and display them again when the page is reloaded. You can store the [`FileSelectFileInfo`]({%slug fileselect-events%}#fileselectfileinfo) records received during the [`OnSelect`]({%slug fileselect-events%}#onselect) event and load them in FileSelect when the page is loaded.
```

Comment:
What exactly are the selected files? Are these files that a user has selected previously? If they are, consider the following suggestion:

```suggestion
The Initial Files feature of the FileSelect allows you to save a list of files that the user has selected. Then, you can display them again when the page is reloaded. To achieve this:
1. Store the [`FileSelectFileInfo`]({%slug fileselect-events%}#fileselectfileinfo) records received during the [`OnSelect`]({%slug fileselect-events%}#onselect) event.
1. Load the [`FileSelectFileInfo`]({%slug fileselect-events%}#fileselectfileinfo) records in the FileSelect when the page is loaded.
```